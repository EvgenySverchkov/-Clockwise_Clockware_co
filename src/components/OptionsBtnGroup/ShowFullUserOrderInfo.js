import React from "react";
import { useDispatch } from "react-redux";
import { chngCurrItemForModal } from "../../store/adminModalWindows/actions";

const ShowFullUserOrderInfo = ({ itemObj }) => {
  const dispatch = useDispatch();
  return (
    <td className="text-center">
      <button
        onClick={() => {
          dispatch(chngCurrItemForModal(itemObj));
        }}
        type="button"
        className="btn btn-secondary"
        data-toggle="modal"
        data-target="#fullInfoModal"
      >
        Show full info
      </button>
    </td>
  );
};

export default ShowFullUserOrderInfo;
