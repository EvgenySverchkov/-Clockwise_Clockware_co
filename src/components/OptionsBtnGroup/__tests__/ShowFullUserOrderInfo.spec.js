import React from "react";
import {create} from "react-test-renderer";
import {Provider} from "react-redux";
import configMockStore from 'redux-mock-store';

import ShowFullUserOrderInfo from "../ShowFullUserOrderInfo";

const mockStore = configMockStore();

it("Test of <ShowFullUserOrderInfo />", ()=>{
    const store = mockStore({})
    const mockProps = {
        itemObj: {
            someField: "Some Value"
        }
    }
    const component = create(
        <Provider store = {store}>
            <ShowFullUserOrderInfo {...mockProps}/>
        </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
});