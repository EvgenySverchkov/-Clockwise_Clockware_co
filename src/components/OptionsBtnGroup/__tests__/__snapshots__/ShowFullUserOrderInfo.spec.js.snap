// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <ShowFullUserOrderInfo /> 1`] = `
<td
  className="text-center"
>
  <button
    className="btn btn-secondary"
    data-target="#fullInfoModal"
    data-toggle="modal"
    onClick={[Function]}
    type="button"
  >
    Show full info
  </button>
</td>
`;
